custom_components/tahoma/cover.py

```diff
@@ -258,16 +258,58 @@ async def async_close_cover_tilt(self, **_):
 
     async def async_stop_cover(self, **_):
         """Stop the cover."""
-        await self.async_execute_command(
-            self.select_command(COMMAND_STOP, COMMAND_STOP_IDENTIFY, COMMAND_MY)
+        exec_id = next(
+            (
+                exec_id
+                for exec_id, execution in self.coordinator.executions.items()
+                if execution.get("deviceurl") == self.device.deviceurl
+                and execution.get("command_name")
+                in [
+                    COMMAND_OPEN,
+                    COMMAND_UP,
+                    COMMAND_CYCLE,
+                    COMMAND_CLOSE,
+                    COMMAND_DOWN,
+                    COMMAND_SET_POSITION,
+                    COMMAND_SET_CLOSURE,
+                    COMMAND_SET_PEDESTRIAN_POSITION,
+                    COMMAND_SET_POSITION_AND_LINEAR_SPEED,
+                ]
+            ),
+            None,
         )
 
+        if exec_id:
+            logging.debug("Cancelling command " + exec_id)
+            await self.async_cancel_command(exec_id)
+        else:
+            logging.debug("Calling stop command")
+            await self.async_execute_command(
+                self.select_command(COMMAND_STOP, COMMAND_STOP_IDENTIFY, COMMAND_MY)
+            )
+
     async def async_stop_cover_tilt(self, **_):
```

Comment:
This method is exactly the same that the above one, except for the command names, you can create a common method.